src/test_helpers/app.rs

```diff
@@ -35,7 +36,8 @@ where
 {
     initialize_telemetry();
 
-    let app = app::build(cfg).await.unwrap();
```

Comment:
Not needed for this PR but I think we should try to split up `app::build` (maybe needs an `AppBuilder` struct) so we can easily override things that can't be set via config. (like manually creating a MockCaClient with bad certs, or custom logic to give bad certs for only one ID). 

`app::new` would be where we set our defaults.

Reply:
I was making an AppBuilder at first but got annoyed so went with this for now :-)
 but I agree